src/main/groovy/com/deploygate/gradle/plugins/dsl/NamedDeployment.java

```diff
@@ -0,0 +1,211 @@
+package com.deploygate.gradle.plugins.dsl;
+
+import com.deploygate.gradle.plugins.dsl.syntax.DeploymentSyntax;
+import com.deploygate.gradle.plugins.internal.DeprecationLogger;
+import groovy.lang.Closure;
+import org.gradle.api.Action;
+import org.gradle.api.Named;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.io.File;
+import java.util.Objects;
+
+/*
+ * This file cannot be written in Groovy until we decide to drop supporting Gradle 6.7 or lower.
+ * Calling Action from such lower Gradle do not work if interfaces are originally written in Groovy.
+ *
+ * For the more details, please check https://github.com/DeployGate/gradle-deploygate-plugin/pull/144 out.
+ */
+public class NamedDeployment implements Named, DeploymentSyntax {
+    @Nonnull
+    private final String name;
+
+    @Nullable
+    private File sourceFile;
+
+    @Nullable
+    private String message;
+
+    private boolean skipAssemble;
+
+    @Nullable
+    @Deprecated
+    private String visibility;
+
+    // Avoid using Optional like Guava for now because we want to reduce external dependencies as much as possible.
+    @Nonnull
+    private Distribution[] optionalDistribution;
+
+    public NamedDeployment(@Nonnull String name) {
+        this.name = name;
+        this.optionalDistribution = new Distribution[]{new Distribution()};
+    }
+
+    @Override
+    @Nonnull
+    public String getName() {
+        return name;
+    }
+
+    public boolean isSkipAssemble() {
+        return skipAssemble;
+    }
+
+    /**
+     * for Kotlin properly access
+     */
+    public boolean getSkipAssemble() {
+        return skipAssemble;
+    }
+
+    @Override
+    public void setSkipAssemble(boolean skipAssemble) {
+        this.skipAssemble = skipAssemble;
+    }
+
+    @Nullable
+    public File getSourceFile() {
+        return sourceFile;
+    }
+
+    @Override
+    public void setSourceFile(@Nullable File sourceFile) {
+        this.sourceFile = sourceFile;
+    }
+
+    @Nullable
+    public String getMessage() {
+        return message;
+    }
+
+    @Override
+    public void setMessage(@Nullable String message) {
+        this.message = message;
+    }
+
+    @Override
+    public void distribution(@Nonnull Action<Distribution> builder) {
+        builder.execute(getDistribution());
+    }
+
+    @Override
+    public void distribution(@Nonnull Closure cl) {
+        Distribution distribution = getDistribution();
+        cl.setDelegate(distribution);
+        cl.setResolveStrategy(Closure.DELEGATE_ONLY);
+        cl.call(distribution);
+    }
+
+    public boolean hasDistribution() {
+        return getDistribution().isPresent();
+    }
+
+    @Nonnull
+    public Distribution getDistribution() {
+        return optionalDistribution[0];
```

Comment:
https://github.com/DeployGate/gradle-deploygate-plugin/pull/144/files#diff-0ac52e1ba8973482bdd7896222e2135ae8c15a3b8336aadff46afc50683a3614L48
The existing code excludes empty.
The code always returns the 0th Distribution.

Is this a not problem in behavior?

Reply:
Totally fine. We can ignore a few rare cases because this field is only for internal use. And also, Groovy has optional chaining but Java not. That's why it's good to convert nullable to non-null.